src/test/testCognicityServer.js

```diff
@@ -0,0 +1,676 @@
+              if (err) {
+                test.fail(err.message + ' ' + JSON.stringify(res));
+              }
+              else {
+                done();
+              }
+           });
+        });
+    });
+
+    // Cards end to end test
+    describe('End-to-end card test', function() {
+
+      let cardId = '0';
+
+      // Request a card, submit and get resulting card details
+      it('Get card one time link', function(done){
+          test.httpAgent(app)
+            .post('/cards')
+            .send({
+                "username": "testuser",
+                "network": "twitter",
+                "language": "en"
+            })
+            .expect(200)
+            .expect('Content-Type', /json/)
+            .end(function(err, res){
+              if (err) {
+                test.fail(err.message + ' ' + JSON.stringify(res));
+              }
+              else {
+                cardId = res.body.cardId
+                test.value(res.body.created).is(true);
+                done();
+              }
+           });
+        });
+
+        // Request a card, submit and get resulting report
+        it('Put card data', function(done){
+            test.httpAgent(app)
+              .put('/cards/'+cardId)
+              .send({
+                  "water_depth": 20,
+                  "text": "big flood",
+                  "created_at": "2017-06-07T07:00:00+0700",
+                  "location": {
+                    "lat": -6.4,
+                    "lng": 106.6
+                  }
+              })
+              .expect(200)
+              .expect('Content-Type', /json/)
+              .end(function(err, res){
+                if (err) {
+                  test.fail(err.message + ' ' + JSON.stringify(res));
+                }
+                else {
+                  console.log(res.body);
+                  done();
+                }
+             });
+          });
+
+          // Request a card, submit and get resulting report
+          it('Put card data', function(done){
+              test.httpAgent(app)
+                .get('/cards/'+cardId)
+                .expect(200)
+                .expect('Content-Type', /json/)
+                .end(function(err, res){
+                  if (err) {
+                    test.fail(err.message + ' ' + JSON.stringify(res));
+                  }
+                  else {
+                    test.value(res.body.result.card_id).is(cardId);
+                    test.value(res.body.result.username).is('testuser');
+                    test.value(res.body.result.network).is('twitter')
+                    test.value(res.body.result.language).is('en')
+                    test.value(res.body.result.report.text).is('big flood')
+                    done();
+                  }
+               });
+            });
+
+            // Update a card
+            it('Update card data', function(done){
+                test.httpAgent(app)
+                  .patch('/cards/'+cardId)
```

Comment:
who's allowed to patch a card? Are we restricting this endpoint through api gateway?